core/src/main/scala/org/apache/spark/ui/WebUI.scala

```diff
@@ -134,7 +134,8 @@ private[spark] abstract class WebUI(
   def bind() {
     assert(!serverInfo.isDefined, "Attempted to bind %s more than once!".format(className))
     try {
-      serverInfo = Some(startJettyServer("0.0.0.0", port, sslOptions, handlers, conf, name))
+      var host = Option(conf.getenv("SPARK_LOCAL_IP")).getOrElse("0.0.0.0")
+      serverInfo = Some(startJettyServer(host, port, sslOptions, handlers, conf, name))
       logInfo("Started %s at http://%s:%d".format(className, publicHostName, boundPort))
```

Comment:
I think the log line should reflect the host that was bound to, even if it's 0.0.0.0. While you're at it you can make it an interpolated string. I think this is fine to try again otherwise.